src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -505,6 +505,8 @@ public boolean assignCitizen(final ICitizenData citizen)
             colony.getProgressManager()
               .progressEmploy(colony.getCitizenManager().getCitizens().stream().filter(citizenData -> citizenData.getJob() != null).collect(Collectors.toList()).size());
         }
+
+        updateWorkerAvailableForRecipes();
```

Comment:
Maybe have a check for "complex recipes" so that this doesn't happen on private recipe add